pkg/reconciler/revision/revision.go

```diff
@@ -78,18 +78,24 @@ func (c *Reconciler) reconcileDigest(ctx context.Context, rev *v1.Revision) erro
 		ServiceAccountName: rev.Spec.ServiceAccountName,
 		ImagePullSecrets:   imagePullSecrets,
 	}
-	digest, err := c.resolver.Resolve(rev.Spec.GetContainer().Image,
-		opt, cfgs.Deployment.RegistriesSkippingTagResolving)
-	if err != nil {
-		err = fmt.Errorf("failed to resolve image to digest: %w", err)
-		rev.Status.MarkContainerHealthyFalse(v1.ReasonContainerMissing,
-			v1.RevisionContainerMissingMessage(
-				rev.Spec.GetContainer().Image, err.Error()))
-		return err
+	if rev.Status.ImageDigests == nil {
+		rev.Status.ImageDigests = make(map[string]string, len(rev.Spec.Containers))
+	}
+	for _, container := range rev.Spec.Containers {
+		digest, err := c.resolver.Resolve(container.Image,
```

Comment:
Do we want to do this asyncronously?
I remember @markusthoemmes having problem with each taking 10s?

Reply:
na, my issues was entirely separate. This is fine usually. I suppose we could run them in parallel though?

Reply:
yeah, that's what I meant in go routines.

Reply:
Modified Thank you @markusthoemmes @vagababov